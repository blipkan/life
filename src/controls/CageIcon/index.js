import React from 'react'
import * as PropTypes from 'prop-types'

import { createCompose } from '../../utils/compose'

import Icon from '../common/Icon'

import './index.scss'

export const Defaults = {
  className: 'cage-icon',
  domId: 'cage-icon'
}

const CageIcon = props => {
  const { iconKey } = props

  const compose = createCompose(props, Defaults)

  return (
    <Icon
      className={compose.classNameAndModifier(iconKey)}
      domId={compose.id(iconKey)}
      iconKey={iconKey}
    />
  )
}

CageIcon.propTypes = {
  className: PropTypes.string,
  domId: PropTypes.string,
  iconKey: PropTypes.string
}

export default CageIcon
