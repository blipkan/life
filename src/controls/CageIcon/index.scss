.cage-icon {
  display: flex;
  border-radius: 50%;
  width: 42px;
  height: 42px;
  justify-content: center;
  align-items: center;
  user-select: none;
  color: #000;
  margin: 16px;

  img {
    width: 20px;
    height: 20px;
  }

  &_dead {
    background: rgb(30, 117, 143);
    background: linear-gradient(
      180deg,
      rgba(30, 117, 143, 1) 27%,
      rgba(30, 143, 47, 0.7819502801120448) 96%
    );
  }

  &_live {
    background: rgb(255, 215, 88);
    background: linear-gradient(
      180deg,
      rgba(255, 215, 88, 1) 26%,
      rgba(246, 247, 151, 0.9051995798319328) 75%
    );
  }

  &_birth {
    background: rgb(213, 85, 245);
    background: linear-gradient(
      180deg,
      rgba(213, 85, 245, 1) 45%,
      rgba(247, 151, 224, 0.9668242296918768) 84%
    );
  }
}
