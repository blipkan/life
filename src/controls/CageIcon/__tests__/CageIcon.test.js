/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'

import CageIcon from '../index'

describe('CageIcon', () => {
  const DOMID = 'DOMID'
  const CLASSNAME = 'CLASSNAME'
  const onClick = jest.fn(e => e)

  const Properties = {
    className: CLASSNAME,
    domId: DOMID,
    onClick
  }

  it('render component', () => {
    const component = shallow(<CageIcon />)
    expect(component).toMatchSnapshot()
  })

  it('render props', () => {
    const props = { ...Properties }
    const component = shallow(<CageIcon {...props} />)
    expect(component).toMatchSnapshot()
  })
})
