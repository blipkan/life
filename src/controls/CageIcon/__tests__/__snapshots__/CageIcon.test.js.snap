// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CageIcon render component 1`] = `
<Icon
  className="cage-icon cage-icon"
  domId="cage-icon"
/>
`;

exports[`CageIcon render props 1`] = `
<Icon
  className="cage-icon CLASSNAME cage-icon CLASSNAME"
  domId="DOMID"
/>
`;
